File: frontend/src/types/prediction.ts
export interface PredictionRequest {
  location: string;
  transaction: string;
  furnishing: string;
  facing?: string | null;
  ownership?: string | null;
  carpet_area_sqft: number;
  bathroom: number;
  balcony: number;
  car_parking: number;
  floor_num: number;
}

export interface PredictionResponse {
  predicted_price: number;
}

File: backend/app/schemas/prediction.py
from pydantic import BaseModel


class PredictionRequest(BaseModel):
    location: str
    transaction: str
    furnishing: str
    facing: str | None = None
    ownership: str | None = None
    carpet_area_sqft: float
    bathroom: float
    balcony: float
    car_parking: float
    floor_num: float


class PredictionResponse(BaseModel):
    predicted_price: float